addons/purchase/models/purchase_order.py

```diff
@@ -814,6 +816,80 @@ def _get_orders_to_remind(self):
             ('mail_reminder_confirmed', '=', False)
         ]).filtered(lambda p: p.mapped('order_line.product_id.product_tmpl_id.type') != ['service'])
 
+    def _get_action_add_from_catalog_extra_context(self):
+        return {
+            **super()._get_action_add_from_catalog_extra_context(),
+            'display_uom': self.env.user.has_group('uom.group_uom'),
+            'precision': self.env['decimal.precision'].precision_get('Product Unit of Measure'),
+            'product_catalog_currency_id': self.currency_id.id,
+            'product_catalog_digits': self.order_line._fields['price_unit'].get_digits(self.env),
+            'search_default_seller_ids': self.partner_id.name,
+        }
+
+    def _get_product_catalog_domain(self):
+        return expression.AND([super()._get_product_catalog_domain, [('purchase_ok', '=', True)]])
+
+    def _get_product_catalog_order_data(self, products, **kwargs):
+        return {product.id: self._get_product_price_and_data(product) for product in products}
+
+    def _get_product_catalog_record_lines(self, product_ids):
+        grouped_lines = defaultdict(lambda: self.env['purchase.order.line'])
+        for line in self.order_line:
+            if line.display_type or line.product_id.id not in product_ids:
+                continue
+            grouped_lines[line.product_id] |= line
+        return grouped_lines
+
+    def _get_product_price_and_data(self, product):
+        """ Fetch the product's data used by the purchase's catalog.
+
+        :return: the product's price and the minimum quantity to buy and the
+                 product's packaging data is appliant
```

Comment:
product's packaging data is appliant => product's packaging data if applicable

I think is what you mean